File: cems.API/Features/LogBrowser/Dtos/GroupListItemDto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using cems.API.Models;

namespace cems.API.Features.LogBrowser.Dtos
{
    public class GroupListItemDto
    {
        public int Id { get; set; }
        public GroupingReason GroupingReason { get; set; }
        public string GroupingContext { get; set; }
        public DateTime FirstOccured { get; set; }
        public DateTime LastOccured { get; set; }
        public int LogsCount { get; set; }

        public LogHeaderDto LastLogHeader { get; set; }

    }
}

File: cems-SPA/src/app/dashboard/models/GroupListItem.dto.ts
import { GroupingReason } from 'src/app/_models/GroupingReason.enum';
import { LogHeader } from './LogHeader.dto';

export class GroupListItem {
    public id: number;
    public groupingReason: GroupingReason;
    public groupingContext: string;
    public firstOccured: Date;
    public lastOccured: Date;
    public logsCount: number;

    public lastLogHeader: LogHeader;

    constructor(groupItemObj) {
        this.id = groupItemObj.id;
        this.groupingReason = groupItemObj.groupingReason;
        this.groupingContext = groupItemObj.groupingContext;
        this.firstOccured = groupItemObj.firstOccured;
        this.lastOccured = groupItemObj.lastOccured;
        this.logsCount = groupItemObj.logsCount;

        this.lastLogHeader = new LogHeader(groupItemObj.lastLogHeader);
    }


}
